### DISCUSSION TITLE
Using Yetiforce functions globally

### DISCUSSION THREAD

**Discussion by drakub** [YetiForceCompany/YetiForceCRM#14455] [General]
I am trying to create an independent application as a portal. But I wanted to register portal users using yetiforce function for registering users and also do the same for changing password. I am facing challenges, I tried to follow the code but it doesn't help me either. I tried using the reset password script created only to realize that this doesn't help either. But I would really like to use the PasswordReset script but only wrap the logic in a function and pass two arguments to this function (userid, password). Can someone help me on how I can do this or walk me through the steps of generating these passwords exactly the way yetiforce has done but without the CRM application or direct me step by step on how the CRM has implemented it on Users Module.

Regards,

**Comment by hahyes** [comment]
> I am trying to create an independent application as a portal. But I wanted to register portal users using yetiforce function for registering users and also do the same for changing password. I am facing challenges, I tried to follow the code but it doesn't help me either. I tried using the reset password script created only to realize that this doesn't help either. But I would really like to use the PasswordReset script but only wrap the logic in a function and pass two arguments to this function (userid, password). Can someone help me on how I can do this or walk me through the steps of generating these passwords exactly the way yetiforce has done but without the CRM application or direct me step by step on how the CRM has implemented it on Users Module.
> 
> Regards,

Did you try using YetiForce API to do that? I'm not sure that creating users is included, but this is for sure better and easier option.

**Reply by drakub** [reply]
Yetiforce does not have those APIs for doing that, however, I have managed to move around with user creation. What I have done is to create a function in the PasswordReset.php file and then call this with a user id that I have created to set the password. Works perfect because when I login with the returned password (sent on mail), it works ok.

Now the challenge I have is using the login function of the CRM in my portal. Still trying to figure this out but anyone who gets this faster can get me up and running.

Find attached my function and function call in the custom application. I am using CodeIgniter to do the custom portal application.

**The Password Reset Function**
![resetFunction](https://user-images.githubusercontent.com/19745222/112309360-4e43cc00-8cb4-11eb-8991-e41d0e346895.JPG)

**The Custom Registration Function**
![userRegistration](https://user-images.githubusercontent.com/19745222/112309796-d6c26c80-8cb4-11eb-8011-c373d28874e2.JPG)

**Reply by hahyes** [reply]
> Yetiforce does not have those APIs for doing that

https://github.com/YetiForceCompany/YetiForceCRM/issues/11770
https://doc.yetiforce.com/api/

**Reply by drakub** [reply]
I know there are APIs of yetiforce for other things but not this custom login. What is available would involve a lot of manual work
